Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -270,88 +239,92 @@ function Add-AzureStackDependencyData {
             $Endpoint.Data.AzureKeyVaultDnsSuffix = $AzureKeyVaultDnsSuffix
         }
     } 
-    else 
-    {
+    else {
         throw "Unable to fetch Azure Stack Dependency Data."
     }
     return $Endpoint
 }
 
 function Has-ObjectProperty {
     [CmdletBinding()]
-    param([Parameter(Mandatory=$true)] $object,
-    [Parameter(Mandatory=$true)] $propertyName) 
+    param([Parameter(Mandatory = $true)] $object,
+        [Parameter(Mandatory = $true)] $propertyName) 
 
-    if(Get-Member -inputobject $object -name $propertyName -Membertype Properties)
-    {
+    if (Get-Member -inputobject $object -name $propertyName -Membertype Properties) {
         return $true
     }
-    else
-    {
+    else {
         return $false
     }
 }
 
 function Get-AzureRMAccessToken {
     [CmdletBinding()]
-    param([Parameter(Mandatory=$true)] $endpoint) 
+    param(
+        [Parameter(Mandatory = $true)] $endpoint,
+        [parameter(Mandatory = $false)] $overrideResourceType = $null
+    ) 
 
-    if($endpoint.Auth.Scheme -eq $MsiConnection )
-    {
+    if ($endpoint.Auth.Scheme -eq $MsiConnection ) {
         Get-MsiAccessToken $endpoint 0 0
     }
-    else
-    {
+    else {
         if ($Endpoint.Auth.Parameters.AuthenticationType -eq 'SPNCertificate') {
             Get-SpnAccessTokenUsingCertificate $endpoint
```

Comment:
you would also need to add overrideResourceType to `Get-SpnAccessTokenUsingCertificate` and `Get-MsiAccessToken` methods as well.

Reply:
this [commit](https://github.com/microsoft/azure-pipelines-tasks/pull/13770/commits/33b987a9d3b50e5ef74cd193ddd33f69f62ec2c2) achieves this. Deployed a copy of the task and tested it live